scipy/optimize/_optimize.py

```diff
@@ -131,6 +132,45 @@ def _dict_formatter(d, n=0, mplus=1, sorter=None):
     return s
 
 
+def _wrap_callback(callback):
```

Comment:
If you send the method to `_wrap_callback` then we won't need to figure out whether it's trust-constr based on the length of signature.

Reply:
I figured we might as well use the information we were already looking at. But if you prefer, I'll change it.